modin/engines/base/frame/data.py

```diff
@@ -1996,26 +1996,50 @@ def _concat(self, axis, others, how, sort):
         )
 
     def groupby_reduce(
-        self, axis, by, map_func, reduce_func, new_index=None, new_columns=None
+        self,
+        axis,
+        by,
+        map_func,
+        reduce_func,
+        new_index=None,
+        new_columns=None,
+        apply_indices=None,
     ):
         """Groupby another dataframe and aggregate the result.
 
-        Args:
-            axis: The axis to groupby and aggregate over.
-            by: The dataframe to group by.
-            map_func: The map component of the aggregation.
-            reduce_func: The reduce component of the aggregation.
-            new_index: (optional) The index of the result. We may know this in advance,
+        Parameters
+        ----------
+            axis: int,
+                The axis to groupby and aggregate over.
+            by: ModinFrame (optional),
+                The dataframe to group by.
+            map_func: callable,
+                The map component of the aggregation.
+            reduce_func: callable,
+                The reduce component of the aggregation.
+            new_index: Index (optional),
+                The index of the result. We may know this in advance,
                 and if not provided it must be computed.
-            new_columns: (optional) The columns of the result. We may know this in
-                advance, and if not provided it must be computed.
+            new_columns: Index (optional),
+                The columns of the result. We may know this in advance,
+                and if not provided it must be computed.
+            apply_indices : list-like (optional),
+                Indices of `axis ^ 1` to apply groupby over.
 
         Returns
         -------
              A new dataframe.
         """
+        by_parts = by if by is None else by._partitions
+
+        if apply_indices is not None:
```

Comment:
Where is the code that initially defines `apply_indices`?

All the code for that variable is written in context - if the variable is defined ...

Reply:
if `apply_indices is None` then we pass `None` to the partition manager, and it [handles it](https://github.com/dchigarev/modin/blob/830d2ea30f1a9d3c1cd7ca0f1ee9d43a04831ff4/modin/engines/base/frame/partition_manager.py#L127) correctly

Reply:
How it handles it if condition:
```python
        if apply_indices is not None:
```

Reply:
I will rephrase where in the code `apply_indices` gets a value differ from `None` ?

Reply:
[here](https://github.com/dchigarev/modin/blob/830d2ea30f1a9d3c1cd7ca0f1ee9d43a04831ff4/modin/data_management/functions/groupby_function.py#L146-L150), we're specifying `apply_indices` only when we're doing dictionary aggregation, it helps us to do not touch those partitions, that doesn't contain columns to aggregate. In all other cases we're applying groupby to the whole frame and `apply_indices is None`

Reply:
Thanks, for some reason I didn't see the place.